app.go

```diff
@@ -524,10 +526,14 @@ func (a *Application) BeginBlock(req abci.RequestBeginBlock) abci.ResponseBeginB
 			panic(err)
 		}
 
-		if err := chainConfigManager.UpdateConfig(); err != nil {
+		updatedConfig, err := chainConfigManager.UpdateConfig()
+		if err != nil {
 			panic(err)
 		}
 
+		if updatedConfig > 0 {
+			a.config = loadOnChainConfig(state)
+		}
```

Comment:
I'd just do:
```golang
if numConfigChanges > 0 {
  // invalidate cached config so it's reloaded next time it's accessed
  a.config = nil
}
```

Reply:
Yes, that would be better in term of performance I think